### PR TITLE
feat: router sınıflarına controller yapısı eklendi

### PR DESCRIPTION
**Pull request by VeyselUstuntas**

### CODE REVIEW

File: index.php [hunk 1]
```diff
@@ -1,40 +1,21 @@
...
-    )
-);
+$prime = new PrimeController(PrimeRepository::repositoryContext());
+Router::register($prime->prime());
 
-$router->register(new Route('home', function () {
-    require 'home.php';
-}));
+$home = new HomeController();
+Router::register($home->home());
```
Comment (yigitsahinupwork) [inline index.php:19]: Where is the method for the route?

File: index.php [hunk 2]
```diff
@@ -1,40 +1,21 @@
...
-            $primeNumber = new PrimeNumber();
-            $primeNumber->setPrimeNumberLimit($parameter);
-            $result = $primeNumber->stringify();
-            $title = "Prime Numbers";
-            require "show.php";
-        }
-    )
-);
+$prime = new PrimeController(PrimeRepository::repositoryContext());
+Router::register($prime->prime());
```
Comment (yigitahmetsahin) [inline index.php:16]: Where is the method for the route?

File: controller/fibonacci-controller.php [hunk 3]
```diff
@@ -0,0 +1,23 @@
+<?php
+class FibonacciController
+{
+    private $_context;
```
Comment (yigitahmetsahin) [inline controller/fibonacci-controller.php:4]: What is the type of the context?

File: controller/fibonacci-controller.php [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+
+    public function __construct($context)
+    {
+        $this->_context = $context;
+    }
+
+    public function fibonacci()
+    {
+        if ($_SERVER["REQUEST_METHOD"] === "POST") {
```
Comment (yigitahmetsahin) [inline controller/fibonacci-controller.php:13]: Why do we have http method handlers here?

**Pull request merged**